geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/GeoMesaCoprocessor.scala

```diff
@@ -0,0 +1,154 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.hbase.coprocessor
+
+import java.io._
+
+import com.google.protobuf.{ByteString, RpcCallback, RpcController, Service}
+import org.apache.hadoop.hbase.client.coprocessor.Batch.Call
+import org.apache.hadoop.hbase.client.{Scan, Table}
+import org.apache.hadoop.hbase.coprocessor.{CoprocessorException, CoprocessorService, RegionCoprocessorEnvironment}
+import org.apache.hadoop.hbase.exceptions.DeserializationException
+import org.apache.hadoop.hbase.filter.{FilterList, Filter => HFilter}
+import org.apache.hadoop.hbase.ipc.BlockingRpcCallback
+import org.apache.hadoop.hbase.protobuf.ResponseConverter
+import org.apache.hadoop.hbase.{Cell, Coprocessor, CoprocessorEnvironment}
+import org.locationtech.geomesa.features.SerializationOption.SerializationOptions
+import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer
+import org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor._
+import org.locationtech.geomesa.hbase.coprocessor.aggregators.GeoMesaHBaseAggregator
+import org.locationtech.geomesa.hbase.coprocessor.utils.{GeoMesaHBaseCallBack, GeoMesaHBaseRpcController}
+import org.locationtech.geomesa.hbase.proto.GeoMesaProto
+import org.locationtech.geomesa.hbase.proto.GeoMesaProto.{GeoMesaCoprocessorRequest, GeoMesaCoprocessorResponse, GeoMesaCoprocessorService}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.io.CloseQuietly
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+class GeoMesaCoprocessor extends GeoMesaCoprocessorService with Coprocessor with CoprocessorService {
+
+  private var env: RegionCoprocessorEnvironment = _
+
+  @throws[IOException]
+  def start(env: CoprocessorEnvironment): Unit = {
```

Comment:
can you add 'override' markers to the methods that are overridden? makes it a lot easier to reason about things